### PR TITLE
Fix the no-source bug in gdrdrv-dkms.deb (issue #131)

### PR DESCRIPTION
**Pull request by pakmarkthub**
Problems:
- See issue #131. 
- The dkms build scripts incorrectly handle DEBIAN\_VERSION (root cause).
- In build-deb-packages.sh, dpkg-buildpackage incorrectly marks dkms as binary-only package. This is the root cause of issue #131 but should be fixed nonetheless.

This PR:
- fixes issue #131.
- modifies build-deb-packages.sh and internal scripts to correctly handle DEBIAN\_VERSION.
- upgrades dkms source version to 3.0.

Pre-submit testing:
- On Ubuntu18.04.
- Tested install and purge for both gdrcopy.deb and gdrdrv-dkms.deb.
- Tested upgrade gdrdrv-dkms from version 2.0 to 2.1.

### CODE REVIEW

File: packages/build-deb-packages.sh [hunk 1]
```diff
@@ -41,56 +41,60 @@ if [ "X$VERSION" == "X" ]; then
...
 
 echo
 echo "Building dkms module ..."
-ex cd $tmpdir/gdrcopy-${VERSION}/src/gdrdrv
+ex cd ${tmpdir}/gdrcopy-${VERSION}/src/gdrdrv
 ex make clean
 
-ex mkdir -p $tmpdir/gdrdrv-dkms-${VERSION}/
-ex cp -r $tmpdir/gdrcopy-${VERSION}/src/gdrdrv $tmpdir/gdrdrv-dkms-${VERSION}/gdrdrv-${VERSION}
-ex cp ${SCRIPT_DIR_PATH}/dkms.conf $tmpdir/gdrdrv-dkms-${VERSION}/gdrdrv-${VERSION}/
```
Comment (drossetti) [inline packages/build-deb-packages.sh:80]: I argue that the driver source code must not change when simply changing DEBIAN_VERSION, hence the dir should only versioned with MAJOR.MINOR

File: packages/build-deb-packages.sh [hunk 2]
```diff
@@ -41,56 +41,60 @@ if [ "X$VERSION" == "X" ]; then
...
 fi
 
 echo "Building gdrcopy debian packages version ${FULL_VERSION} ..."
 
-echo "Working in $tmpdir ..."
+echo "Working in ${tmpdir} ..."
 
 ex cd ${TOP_DIR_PATH}
 
-ex mkdir -p $tmpdir/gdrcopy
```
Comment (drossetti) [inline packages/build-deb-packages.sh:56]: can you separate the ${tmpdir} cleanup in a separate patch within this same PR ?
it makes review much easier

**Comment by pakmarkthub** [comment]
@drossetti 

This should fix as you suggested. Please re-review.

By the way, since you plan to create a different PR, I didn't add gdrcopy_ prefix to gdrdrv-dkms.deb here.

File: packages/build-deb-packages.sh [hunk 3]
```diff
@@ -42,55 +42,65 @@ fi
...
-ex cd $tmpdir/gdrdrv-dkms-${VERSION}/
+dkmsdir="${tmpdir}/gdrdrv-dkms-${VERSION}"
+ex mkdir -p ${dkmsdir}
+ex cp -r ${tmpdir}/gdrcopy-${VERSION}/src/gdrdrv ${dkmsdir}/gdrdrv-${VERSION}
+ex cp ${SCRIPT_DIR_PATH}/dkms.conf ${dkmsdir}/gdrdrv-${VERSION}/
+ex cd ${dkmsdir}
 ex cp -r ${SCRIPT_DIR_PATH}/dkms/* .
-ex find . -type f -exec sed -i "s/@VERSION@/${FULL_VERSION}/g" {} +
+ex find . -type f -exec sed -i "s/@FULL_VERSION@/${FULL_VERSION}/g" {} +
+ex find . -type f -exec sed -i "s/@VERSION@/${VERSION}/g" {} +
```
Comment (drossetti) [inline packages/build-deb-packages.sh:86]: dkmsdir is a subdir of tmpdir, are these two 'find + sed' commands superfluous ?

**Comment by drossetti** [comment]
> By the way, since you plan to create a different PR, I didn't add gdrcopy_ prefix to gdrdrv-dkms.deb here.

it makes sense!

File: packages/build-deb-packages.sh (hunk 3 above)
Comment (pakmarkthub) [reply packages/build-deb-packages.sh:86]: No, they are not. These two commands are important in handling deb_version while keep the gdrdrv folder without deb_version.

Note that FULL_VERSION and VERSION refer to different strings.

Comment (drossetti) [reply packages/build-deb-packages.sh:86]: understood.
I referred to lines 64 and 65. But I see that you copy stuff into  ${dkmsdir}/gdrdrv-${VERSION} at line 81, which is later, so you need these extra sed.

**Pull request merged**